src/test/kotlin/org/malachite/estella/queue/TaskResultTest.kt

```diff
@@ -0,0 +1,138 @@
+
+    @Autowired
+    private lateinit var taskStageRepository: TaskStageRepository
+
+
+    private lateinit var task: Task
+    private lateinit var taskStage: TaskStage
+
+    @BeforeEach
+    fun prepareTask() {
+        task = Task(null, SerialBlob("xd".toByteArray()), SerialClob("xd".toCharArray()), "filename", 10)
+        val jobseeker = jobSeekerRepository.findAll().first()
+
+        val stage = recruitmentStageRepository.findAll().first()
+        val application = Application(null,
+                Date(Calendar.getInstance().time.time),
+                ApplicationStatus.IN_PROGRESS,
+                jobseeker,
+                mutableSetOf(),
+                mutableListOf())
+        val savedApplication = applicationRepository.save(application)
+        val applicationStageData = ApplicationStageData(null, stage, savedApplication, null, null)
+        val savedApplicationStageData = applicationStageDataRepository.save(applicationStageData)
+        this.taskStage = TaskStage(null, listOf(), savedApplicationStageData)
+        taskStage = taskStageRepository.save(taskStage)
+        task = taskRepository.save(task)
+    }
+
+    @Test
+    fun `task result consuming`() {
+        expectThat(taskStage.tasksResult.size).isEqualTo(0)
+
+        val taskResult = TaskResult(null, SerialBlob("xd".toByteArray()), SerialClob("xd".toCharArray()), null, null, task, taskStage)
+        publish(taskResult)
+
+        eventually {
+            taskStage = taskStageRepository.findById(taskStage.id!!).get()
+            expectThat(taskStage.tasksResult.size).isEqualTo(1)
+        }
+    }
+
+    @Test
+    fun `task bad message format consuming`() {
+        expectThat(taskStage.tasksResult.size).isEqualTo(0)
+
+        badPublish()
+
+        coolDown {
+            expectThat(taskStage.tasksResult.size).isEqualTo(0)
+        }
+    }
+
+
+    fun publish(result: TaskResult) {
+        try {
+            val resultBody = mapOf(
+                    "results" to String(result.results.binaryStream.readAllBytes()),
+                    "code" to result.code.characterStream.readText(),
+                    "startTime" to result.startTime.toString(),
+                    "endTime" to result.endTime.toString(),
+                    "solverId" to result.taskStage.id!!.toString(),
+                    "taskId" to result.task.id!!.toString()
+            )
+            val msg = Json.encodeToString(resultBody)
+            rabbitTemplate.send("task_result", Message(Base64.getEncoder().encode(msg.toByteArray())
+                    , MessageProperties()))
+        } catch (e: Exception) {
+            println("Couldn't send test message to consumer")
+            throw e
+        }
+    }
+
+    fun badPublish() {
+        try {
+            val resultBody = mapOf(
+                    "results" to "4",
+                    "potatoes" to "yes",
+                    "solverId" to "1"
```

Comment:
Nie wiem, czy by nie zrobić testu na to, że masz wszystko legitnie, ale jakiegoś jednego pola brakuje / jest złego typu. Tak na wszelki wypadek, żebyśmy wiedzieli, co się dzieje, kiedy to ma miejsce.